symbol_exporter/ast_db_populator.py

```diff
@@ -306,8 +307,10 @@ def diff_sort(val):
             sort_arch_ordering.index(arch),
         )
 
-    pkgs_to_inspect = diff(upstream, existing_pkg_dict)
-    sorted_files = sorted(list(pkgs_to_inspect), key=diff_sort)
+    pkgs_to_inspect = list(diff(upstream, existing_pkg_dict))
+    # shuffle so that we don't always get the same pkgs
+    shuffle(pkgs_to_inspect)
+    sorted_files = sorted(pkgs_to_inspect, key=diff_sort)
```

Comment:
How does shuffling and then sorting give us different packages? I assume it has to do with diff_sort, what does diff_sort do? Can't see full file on my phone. 😛

Reply:
`diff_sort` prioritizes pkgs that we don't have symbols for at all and linux/noarch pkgs. Since it isn't a complete sort, it doesn't uniquely order the pkgs so the shuffle is still somewhat intact.

As an example:
```python
a = ['a1', 'b1', 'c1']
b = ['c1', 'a1', 'b1']

sorted(a, key=lambda x: x[1])
>>> ['a1', 'b1', 'c1']

sorted(b, key=lambda x: x[1])
>>> ['c1', 'a1', 'b1']
```

Reply:
got it